ext/standard/string.c

```diff
@@ -3990,7 +3990,8 @@ static zend_long php_str_replace_in_subject(zval *search, zval *replace, zval *s
 		/* For each entry in the search array, get the entry */
 		ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(search), search_entry) {
 			/* Make sure we're dealing with strings. */
-			SEPARATE_ZVAL(search_entry);
+			ZVAL_DEREF(search_entry); // Deal with possibly shared zvals
```

Comment:
minor one, please don't use "//" as comment, use "/*". or maybe remove this moment at all..